src/esphomelib/sensor/tx20_component.cpp

```diff
@@ -0,0 +1,181 @@
+    return;
+  }
+  if (buffer_index <= MAX_BUFFER_SIZE) {
+    buffer[buffer_index] = delay;
+  }
+  spent_time += delay;
+  start_time = now;
+  buffer_index++;
+}
+TX20WindSpeedSensor *TX20Component::get_wind_speed_sensor() const {
+  return this->wind_speed_sensor_;
+}
+TX20WindDirectionSensor *TX20Component::get_wind_direction_sensor() const {
+  return this->wind_direction_sensor_;
+}
+TX20WindDirectionTextSensor *TX20Component::get_wind_direction_text_sensor() const {
+  return this->wind_direction_text_sensor_;
+}
+
+void TX20Component::decodeAndPublish_() {
+  ESP_LOGV(TAG, "DECODE TX20...");
+
+  std::string string_buffer;
+  std::vector<bool> bit_buffer;
+  uint8_t index_to = 0;
+  bool current_bit = true;
+
+  for (int i = 1; i <= buffer_index; i++) {
+    uint8_t repeat = buffer[i] / TX20_BIT_TIME;
+    string_buffer.append(repeat, current_bit ? '1' : '0');
+    bit_buffer.insert(bit_buffer.end(), repeat, current_bit);
+    current_bit = !current_bit;
+  }
+  if (string_buffer.length() < MAX_BUFFER_SIZE) {
+    uint8_t remain = MAX_BUFFER_SIZE - string_buffer.length();
+    string_buffer.append(remain, current_bit ? '1' : '0');
+    bit_buffer.insert(bit_buffer.end(), remain, current_bit);
+  }
+
+  ESP_LOGV(TAG, "Decoded bits %s", string_buffer.c_str());
+
+  uint8_t tx20_sa = 0;
+  uint8_t tx20_sb = 0;
+  uint8_t tx20_sd = 0;
+  uint8_t tx20_se = 0;
+  uint16_t tx20_sc = 0;
+  uint16_t tx20_sf = 0;
+  uint8_t tx20_wind_direction = 0;
+  float tx20_wind_speed_kmh = 0;
+
+  for (int i = 0; i < 40; i++) {
+    uint8_t bit = bit_buffer.at(i);
+    if (i < 5) {
+      // start, inverted
+      tx20_sa = (tx20_sa << 1) | (bit ^ 1);
+    } else if (i < 5 + 4) {
+      // wind dir, inverted
+      tx20_sb = tx20_sb >> 1 | ((bit ^ 1) << 3);
+    } else if (i < 5 + 4 + 12) {
+      // windspeed, inverted
+      tx20_sc = tx20_sc >> 1 | ((bit ^ 1) << 11);
+    } else if (i < 5 + 4 + 12 + 4) {
+      // checksum, inverted
+      tx20_sd = tx20_sd >> 1 | ((bit ^ 1) << 3);
+    } else if (i < 41 + 5 + 4 + 12 + 4 + 4) {
+      // wind dir
+      tx20_se = tx20_se >> 1 | (bit << 3);
+    } else {
+      // windspeed
+      tx20_sf = tx20_sf >> 1 | (bit << 11);
+    }
+  }
+
+  uint8_t chk = (tx20_sb + (tx20_sc & 0xf) + ((tx20_sc >> 4) & 0xf) + ((tx20_sc >> 8) & 0xf));
+  chk &= 0xf;
+
+  if ((chk == tx20_sd) && (tx20_sc < 400)) {  // if checksum seems to be ok and wind speed below 40 m/s
+    tx20_wind_speed_kmh = float(tx20_sc) * 0.36;
+    tx20_wind_direction = tx20_sb;
+    std::string direction = DIRECTIONS.at(tx20_wind_direction);
+    ESP_LOGV(TAG, "WindSpeed %f, WindDirection %d, WindDirection Text %s", tx20_wind_speed_kmh, tx20_wind_direction,
+             direction.c_str());
+    this->wind_direction_sensor_->publish_state(tx20_wind_direction);
+    this->wind_speed_sensor_->publish_state(tx20_wind_speed_kmh);
+    this->wind_direction_text_sensor_->publish_state(direction);
+  } else {
+    ESP_LOGV(TAG, "Incorrect checksum!");
```

Comment:
Let's make this a warning

Reply:
Happens quite a few times -> Added a comment and will keep it as verbose.